### ISSUE TITLE
Failure to install on Jython since 2.12.0

### ISSUE THREAD

**Issue by ppolewicz** [kennethreitz/requests#3769]
```
(jython) r@r-V ~ $ python --version
Jython 2.7.1b3
(jython) r@r-V ~ $ pip install requests==2.11.1
Collecting requests==2.11.1
  Using cached requests-2.11.1-py2.py3-none-any.whl
Installing collected packages: requests
  Found existing installation: requests 2.11.0
    Uninstalling requests-2.11.0:
      Successfully uninstalled requests-2.11.0
Successfully installed requests-2.11.1
(jython) r@r-V ~ $ pip install requests==2.12.0
Collecting requests==2.12.0
  Using cached requests-2.12.0-py2.py3-none-any.whl
Installing collected packages: requests
  Found existing installation: requests 2.11.1
    Uninstalling requests-2.11.1:
      Successfully uninstalled requests-2.11.1
  Rolling back uninstall of requests
Exception:
Traceback (most recent call last):
  File "/home/r/.virtualenvs/jython/Lib/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/home/r/.virtualenvs/jython/Lib/site-packages/pip/commands/install.py", line 338, in run
    requirement_set.install(
  File "/home/r/.virtualenvs/jython/Lib/site-packages/pip/commands/install.py", line 338, in run
    requirement_set.install(
  File "/home/r/.virtualenvs/jython/Lib/site-packages/pip/req/req_set.py", line 780, in install
    requirement.install(
  File "/home/r/.virtualenvs/jython/Lib/site-packages/pip/req/req_install.py", line 851, in install
    self.move_wheel_files(self.source_dir, root=root, prefix=prefix)
  File "/home/r/.virtualenvs/jython/Lib/site-packages/pip/req/req_install.py", line 1057, in move_wheel_files
    move_wheel_files(
  File "/home/r/.virtualenvs/jython/Lib/site-packages/pip/wheel.py", line 272, in move_wheel_files
    compileall.compile_dir(source, force=True, quiet=True)
  File "/home/r/jython/Lib/compileall.py", line 56, in compile_dir
    if not compile_dir(fullname, maxlevels - 1, dfile, force, rx,
  File "/home/r/jython/Lib/compileall.py", line 56, in compile_dir
    if not compile_dir(fullname, maxlevels - 1, dfile, force, rx,
  File "/home/r/jython/Lib/compileall.py", line 56, in compile_dir
    if not compile_dir(fullname, maxlevels - 1, dfile, force, rx,
  File "/home/r/jython/Lib/compileall.py", line 50, in compile_dir
    if not compile_file(fullname, ddir, force, rx, quiet):
  File "/home/r/jython/Lib/compileall.py", line 99, in compile_file
    ok = py_compile.compile(fullname, None, dfile, True)
  File "/home/r/jython/Lib/compileall.py", line 99, in compile_file
    ok = py_compile.compile(fullname, None, dfile, True)
  File "/home/r/jython/Lib/py_compile.py", line 96, in compile
    _py_compile.compile(file, cfile, dfile)
  File "/home/r/jython/Lib/py_compile.py", line 96, in compile
    _py_compile.compile(file, cfile, dfile)
RuntimeException: java.lang.RuntimeException: Method code too large!
(jython) r@r-V ~ $ 
```
This might not be a bug in `requests`, but perhaps it is workaroundable? Has any change been done in 2.12.0 that caused a very big method to be created?

**Comment by sigmavirus24** [comment]
In the future, please search **closed and** open issues before creating new ones that are duplicates.

This is a duplicate of the **open** #3711

**Comment by ppolewicz** [comment]
I found that issue, but it did not contain "RuntimeException: java.lang.RuntimeException: Method code too large!" and it mentioned usage, not installation, with something called `Indna`, which I didn't think I use.

Sorry for the trouble.

**Comment by nateprewitt** [comment]
Hey @ppolewicz, it looks like this is specifically a limitation of how Jython uses the JVM as noted in [this ticket](http://bugs.jython.org/issue1891). I'd ensure you have the latest version of Jython2.7, and attempt to use Requests 2.12.4.

If that doesn't solve it, I'm afraid there's not much we can do at this time since the bug isn't really with Requests.

**Comment by ppolewicz** [comment]
(for people searching for a solution later and finding this)

That ticket says:

> Issue will get solved if the jython version is 2.7 without applying work around(Splitting of method content).

...but my Jython version is 2.7.1b3 and it still fails.

`requests==2.12.4` fail to install for the same reason.
